components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx

```diff
@@ -164,9 +183,27 @@ export const PortfolioAsset = () => {
     }
 
     // If the id length is greater than 15 assumes it's a contractAddress
-    return assetId.length > 15
+    let token = assetId.length > 15
       ? userVisibleTokensInfo.find((token) => token.contractAddress === assetId)
       : userVisibleTokensInfo.find((token) => token.symbol.toLowerCase() === assetId?.toLowerCase())
+
+    if (!token && assetId.length < 15) {
```

Comment:
@kdenhartog  `Array.find()` method can return the item found in the array or `undefined`. This logic is checking if the result is not `undefined` before proceeding.

Reply:
👍🏻 ahh thank you for pointing that out. I missed that when I was originally reviewing this and it looks fine now. As soon as I looked at it on VS Code it was clear what you're doing here as it reminded me of having to do similar type system checks in the past. In that case then why are we setting the `IsTokenSupported` state to true when this check fails? That part confused me since I'd assume that we'd want to set this to false instead for the else branch.